src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -63,25 +74,68 @@ private static int getPlaceholderPosition(@NotNull GoFunctionOrMethodDeclaration
         return position;
       }
     }
+
     return -1;
   }
 
-  private static int getPlaceholders(@Nullable GoExpression argument) {
-    if (argument == null) return -1;
+  @Nullable
+  private static LinkedHashMap<Integer, String> getPlaceholders(@Nullable GoExpression argument) {
+    if (argument == null) return null;
 
     String placeholderText = resolve(argument);
-    if (placeholderText == null) return -1;
+    if (placeholderText == null) return null;
+
+    int placeholderTextLength = placeholderText.length();
+    int placeholderStart = 0;
+    LinkedHashMap<Integer, String> placeholders = new LinkedHashMap<Integer, String>();
+    while (placeholderStart < placeholderTextLength) {
+      ProgressManager.checkCanceled();
+      if (PLACEHOLDER_START_CHAR != placeholderText.charAt(placeholderStart)) {
+        placeholderStart++;
+        continue;
+      }
+      int placeholderEnd = placeholderStart + 1;
+      while (placeholderEnd < placeholderTextLength &&
+             PLACEHOLDER_ALLOWED_CHARS.indexOf(placeholderText.charAt(placeholderEnd)) != -1 &&
+             PLACEHOLDER_END_CHARS.indexOf(placeholderText.charAt(placeholderEnd)) == -1) {
+        placeholderEnd++;
+      }
+      if (placeholderEnd >= placeholderTextLength) break;
+      if (PLACEHOLDER_ALLOWED_CHARS.indexOf(placeholderText.charAt(placeholderEnd)) == -1 &&
+          PLACEHOLDER_END_CHARS.indexOf(placeholderText.charAt(placeholderEnd)) == -1) {
+        placeholderStart = placeholderEnd;
+        placeholderStart++;
+        continue;
+      }
+      String placeholder = placeholderText.substring(placeholderStart, placeholderEnd + 1);
+      if (!isValidPlaceholder(placeholder)) {
+        placeholders.put(placeholderStart, placeholder);
+      }
+      placeholderStart = placeholderEnd;
+      placeholderStart++;
+    }
+
+    return placeholders;
+  }
 
-    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(placeholderText);
-    Matcher countPlaceholders;
+  private static boolean isValidPlaceholder(String placeholder) {
```

Comment:
I've pushed this last night so that I have it. I'll continue working on this. Thanks for feedback. There's much to work on this to make it report all the things correctly (like ` fmt.Printf("%.*s %d %g\n", 3, "hi", 23, 2.3) `, which is part of the go vet tests)